source/PlayerHailPanel.cpp

```diff
@@ -0,0 +1,326 @@
+/* PlayerHailPanel.cpp
+Copyright (c) 2014 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "PlayerHailPanel.h"
+
+#include "text/alignment.hpp"
+#include "Dialog.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "text/Format.h"
+#include "GameData.h"
+#include "Government.h"
+#include "Information.h"
+#include "Interface.h"
+#include "Messages.h"
+#include "Phrase.h"
+#include "Planet.h"
+#include "PlayerInfo.h"
+#include "Politics.h"
+#include "Ship.h"
+#include "Sprite.h"
+#include "StellarObject.h"
+#include "System.h"
+#include "UI.h"
+
+#include <algorithm>
+#include <cmath>
+
+using namespace std;
+
+
+
+PlayerHailPanel::PlayerHailPanel(PlayerInfo &player, const shared_ptr<Ship> &ship,
+		function<void(const Government *)> bribeCallback)
+	: player(player), ship(ship), bribeCallback(bribeCallback)
+{
+	const Government *gov = ship->GetGovernment();
+	if(!ship->Name().empty())
+		header = gov->GetName() + " " + ship->Noun() + " \"" + ship->Name() + "\":";
+	else
+		header = ship->DisplayModelName() + " (" + gov->GetName() + "): ";
+	// Drones are always unpiloted, so they never respond to hails.
+	bool isMute = ship->GetPersonality().IsMute() || (ship->Attributes().Category() == "Drone");
+	hasLanguage = !isMute && (gov->Language().empty() || player.Conditions().Get("language: " + gov->Language()));
+	canAssistPlayer = !ship->CanBeCarried();
+
+	if(isMute)
+		message = "(There is no response to your hail.)";
+	else if(!hasLanguage)
+		message = "(An alien voice says something in a language you do not recognize.)";
+	else if(gov->IsEnemy())
+	{
+		// Enemy ships always show hostile messages.
+		// They either show bribing messages,
```

Comment:
that's on me ig but that's not the case anymore they can show normal msgs now instead of the bribing all the time